rasa/utils/tensorflow/models.py

```diff
@@ -367,180 +432,14 @@ def load(
         # create empty model
         model = cls(*args, **kwargs)
         # need to train on 1 example to build weights of the correct size
-        model.fit(
-            model_data_example,
-            epochs=1,
-            batch_size=1,
-            evaluate_every_num_epochs=0,
-            evaluate_on_num_examples=0,
-            batch_strategy=SEQUENCE,
-            silent=True,  # don't confuse users with training output
-            loading=True,  # don't use tensorboard while loading
-            eager=True,  # no need to build tf graph, eager is faster here
-        )
+        model.compile()
+        model.fit(model_data_example.as_tf_dataset(1), epochs=1, verbose=0)
```

Comment:
we can pass generator or dataset?

Reply:
yes, you can pass all kinds of things :)